### ISSUE TITLE
Mod Manager does not automatically update vanilla mods

### ISSUE THREAD

**Issue by donottellmetonottellyou** [DFHack/dfhack#4799]
I encountered this issue when switching to the adventure mode branch.

So when you get a mod updated in Dwarf Fortress, Mod Manager (possibly rightly) does not automatically update the mod and will use the old version, allowing the user to manually update through Dwarf Fortress's interface.

However, this is an issue when updating Dwarf Fortress itself, as the vanilla mods do not keep old versions around. DFHack will throw an error and fail the mod loading, which is probably not ideal because the user will have to manually sort and reintegrate the vanilla mods.

Is it possible that vanilla mods could be whitelisted for auto-substitution? As in, if the specified version is not available, the latest version will be used instead? A popup could be used to inform the user of this action/possibility.

Alternatively, all mods could instead be automatically updated instead of being pinned to a specific version.

Related to #3746 

TLDR: Mod Manager errors out when updating to new Dwarf Fortress versions with updated vanilla mods.
